static/app/views/insights/common/components/releaseSelector.tsx

```diff
@@ -134,8 +160,77 @@ function LabelDetails(props: LabelDetailsProps) {
   );
 }
 
+function getSortRealesesBy(
+  sort: ReleasesSortByOption,
+  environments: string[]
+): ReleasesSortByOption {
+  // Require 1 environment for date adopted
+  if (sort === ReleasesSortOption.ADOPTION && environments.length !== 1) {
+    return ReleasesSortOption.DATE;
+  }
+
+  const sortExists = Object.keys(SORT_BY_OPTIONS).includes(sort ?? '');
+
+  if (sortExists) {
+    return sort;
+  }
+
+  // We could give a visual feedback to the user, saying that the sort by is invalid but
+  // since this UI will be refactored, maybe we just don't do anything now
+  return ReleasesSortOption.DATE;
+}
+
 export function ReleaseComparisonSelector() {
   const {primaryRelease, secondaryRelease} = useReleaseSelection();
+  const location = useLocation();
+  const navigate = useNavigate();
+  const {selection} = usePageFilters();
+
+  const [localStoragedReleaseBy, setLocalStoragedReleaseBy] =
+    useLocalStorageState<ReleasesSortByOption>(
+      'insightsReleasesSortBy',
+      ReleasesSortOption.DATE
+    );
+
+  const urlStoragedReleaseBy = decodeScalar(
+    location.query.sortReleasesBy
+  ) as ReleasesSortByOption;
```

Comment:
Users have the option to modify the URL, though I don't expect many to do so. However, if they do, we have a fallback in place: https://github.com/getsentry/sentry/blob/e2ec015b5c245ccdd18e2d6f7afce1402eee0062/static/app/views/insights/common/components/releaseSelector.tsx#L171-L173 The casting here is just to satisfy TypeScript. I do not think we want to constantly trigger an error if users change the URL param, do we?

Reply:
Yes they can, which is why it's good to validate :) You can always just logMessage instead of throw and error btw, because it should still be unexpected behavior